.NET/Microsoft.Recognizers.Text.DateTime/Parsers/BaseMergedDateTimeParser.cs

```diff
@@ -738,28 +738,50 @@ public void AddAltPeriodToResolution(Dictionary<string, string> resolutionDic, s
 
             if (!string.IsNullOrEmpty(mod))
             {
-                // For the 'before' mod, the start of the period should be the end the new period, not the start 
+
+
+                // For the 'before' mod
+                // 1. Cases like "Before December", the start of the period should be the end of the new period, not the start
+                // 2. Cases like "More than 3 days before today", the date point should be the end of the new period
                 if (mod.Equals(Constants.BEFORE_MOD))
                 {
-                    res.Add(DateTimeResolutionKey.END, start);
+                    if (!string.IsNullOrEmpty(start) && !string.IsNullOrEmpty(end))
+                    {
+                        res.Add(DateTimeResolutionKey.END, start);
+                    }
+                    else
```

Comment:
Cases like "Within 2 days after today", "Less than 2 weeks after today".
"Within", "Less than" + datepoint. Datepoint don't have "start" and "end" at the same time before.